rust/src/evmc.rs

```diff
@@ -34,10 +34,10 @@ impl EvmcVm for EvmRs {
             // If this is not the case it violates the EVMC spec and is an irrecoverable error.
             process::abort();
         };
-        let run_state = RunState::new(revision, message, context, code);
-        interpreter::run(run_state)
-            .unwrap_or_else(Into::into)
-            .into()
+        Interpreter::new(revision, message, context, code)
+            .run()
+            .map(Into::into)
+            .unwrap_or_else(|status_code| ExecutionResult::new(status_code, 0, 0, None))
```

Comment:
I got quite confused the other day by this statement. But I get it now. 
The problem that I see here is that `Run` returns a type `Result<Self, StatusCode>` and `StatusCode` is odd type to be the error type since it can be a successful value.

This is not very rustastic, since the purpose of Result is to have Success/Error as disjoint types, and status_code is a union of both cases. 

The problem comes from evmc, the interpretation they do of all those types comes from a naive translation of the C types. I guess this is a necessary evil. But we may want to consider in the future having disjoint types inside the rust vm